src/Compilers/CSharp/Test/Semantic/FlowAnalysis/FlowDiagnosticTests.cs

```diff
@@ -2457,5 +2457,50 @@ public static void Main()
                 Diagnostic(ErrorCode.ERR_UseDefViolation, "k").WithArguments("k").WithLocation(10, 20)
                 );
         }
+
+        [WorkItem(9581, "https://github.com/dotnet/roslyn/issues/9581")]
+        [Fact]
+        public void UsingSelfAssignment()
+        {
+            var source =
+@"class Program
+{
+    static void Main()
+    {
+        using (System.IDisposable x = x)
```

Comment:
Sure, but that won't trigger any of the changed code. That would be a test of the local inference machinery.

---
In reply to: [79030042](https://github.com/dotnet/roslyn/pull/13807#discussion_r79030042) [](ancestors = 79030042)